tests/functional/TestErrorHandling/tests/timeout_test.go

```diff
@@ -0,0 +1,127 @@
+	removeDelayFile := fmt.Sprintf("rm -f %s", delayFile)
+	removePIDFile := fmt.Sprintf("rm -f %s", pidFile)
+
+	cmd := []string{removeSleep, removeDelayFile, removePIDFile}
+	err := rex.AnyError(exec.ExecCommands(node, cmd, 10, true))
+	return err
+}
+
+func cleanupOp(exec *ssh.SshExec, t *testing.T) {
+	for i := 0; i < len(testCluster.Nodes); i++ {
+		sshHost := testCluster.SshHost(i)
+		err := cleanDelay(sshHost, exec)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	}
+}
+
+func TestGlusterTimeout(t *testing.T) {
+	heketiServer := testutils.NewServerCtlFromEnv("..")
+	testutils.ServerStarted(t, heketiServer)
+	defer testutils.ServerStopped(t, heketiServer)
+
+	testCluster.Teardown(t)
+	testCluster.Setup(t, len(testCluster.Nodes), len(testCluster.Disks))
+	defer testCluster.Teardown(t)
+
+	na := testutils.RequireNodeAccess(t)
+	exec := na.Use(logger)
+	//Create a volume
+	volReq := &api.VolumeCreateRequest{}
+	volReq.Size = 1
+	volReq.Snapshot.Enable = true
+	volReq.Snapshot.Factor = 1.5
+	volReq.Durability.Type = api.DurabilityReplicate
+	volReq.Durability.Replicate.Replica = 3
+
+	t.Run("deleteSucceeds", func(t *testing.T) {
+		for i := 0; i < len(testCluster.Nodes); i++ {
+			//61s+61s =2.02 minutes to ensure we are not hitting
+			//default gluster timeout of 2 minutes
+			//this  adds 61s sleep on initiator node and 61s parallel sleep
+			//on remaining nodes
+			err := addDelay(61, testCluster.SshHost(i), exec)
+			tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+		}
+		simplevol, err := heketi.VolumeCreate(volReq)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+		// Delete volume
+		err = heketi.VolumeDelete(simplevol.Id)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+		//remove pid,delay file and clean up sleep process if present
+		defer cleanupOp(exec, t)
+	})
+
+	t.Run("deleteFails", func(t *testing.T) {
+		origConf := path.Join(heketiServer.ServerDir, heketiServer.ConfPath)
+		heketiServer.ConfPath = tests.Tempfile()
+		defer os.Remove(heketiServer.ConfPath)
+		err := UpdateConfig(origConf, heketiServer.ConfPath, func(c *config.Config) {
+			//gluster cli timeout set to 5 seconds
+			c.GlusterFS.SshConfig.GlusterCliTimeout = 5
+		})
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+		heketiServer.KeepDB = true
+		testutils.ServerRestarted(t, heketiServer)
+
+		simplevol, err := heketi.VolumeCreate(volReq)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+		//volume deletion should fail as we have the sleep of
+		//20s+20s=40s which is greater than gluster  cli timeout of 5s
+		for i := 0; i < len(testCluster.Nodes); i++ {
+			//having extra sleep doesnot affect  the testing
+			err := addDelay(20, testCluster.SshHost(i), exec)
+			tests.Assert(t, err == nil, "expected err == nil, got:", err)
+		}
+		// Delete volume
+		err = heketi.VolumeDelete(simplevol.Id)
+		tests.Assert(t, err != nil)
+		defer cleanupOp(exec, t)
```

Comment:
defer line needs to be moved up before the volume create (right after the loop that sets up the delay) so that it executes regardless of the state of the assert